app/services/google/scrape_service.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Google
+  class ScrapeService
+    def initialize(keyword_id:, search_service: Google::SearchService.new)
+      @keyword_id = keyword_id
+      @search_service = search_service
+    end
+
+    def call!
+      keyword = Keyword.find_by id: keyword_id
+      raise_keyword_not_found unless keyword
+      @keyword = keyword
+
+      begin
+        html = search_service.search! keyword.content
```

Comment:
how about move this into a separate method

```
html_response = search

private

def search
  search_service.search! keyword.content
rescue GoogleScraperRuby::Errors::SearchServiceError => error
        raise_unexpected_error error
end
```

Reply:
Sure 22323d5506c9b7c5b6d0f0ff979c9b97fac3df4b